pyth-sdk/src/price.rs

```diff
@@ -346,10 +609,55 @@ impl Price {
             (x as u64, 1)
         }
     }
+
+    /// Helper function to check if precision implied by exponent >= value, exponent <= 0
+    fn check_precision_geq(value: u64, exponent: i32) -> Option<bool> {  
+        if exponent > 0 {
+            return None;
+        }
+        
+        let mut precision: u64 = 1;
+        let mut delta = -exponent;
+        
+        while delta > 0 && (value != 0) {
+            precision = precision.checked_mul(10)?;
+            delta = delta.checked_sub(1)?;
+        }
+
+        return Some(value <= precision);
+    }
+
+    /// Helper function to create fraction and possibly scale to expo -9
+    fn fraction(x: i64, y: i64, scale_expo: Option<i32>) -> Option<Price> {
+        // convert x and y to Prices
+        let x_as_price = Price {
+            price: x,
+            conf: 0,
+            expo: 0,
+            publish_time: 0,
+        };
+        let y_as_price = Price {
+            price: y,
+            conf: 0,
+            expo: 0,
+            publish_time: 0,
+        };
+
+        // get the relevant fraction
+        let mut frac = x_as_price.div(&y_as_price)?;
+
+        if scale_expo.is_some() {
+            frac = frac.scale_to_exponent(scale_expo?)?;
+        }
```

Comment:
As an aside: the way you've defined this function doesn't make sense. Why would you take an Option argument for the exponent and then scale_to_exponent here? You could simply omit the Option and if statement and let the caller call `scale_to_exponent` on the result if they want. That gives you the same functionality with simpler function signatures.